cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -163,7 +163,7 @@ func (kc *kubeletConfig) Default(cfg *kubeadmapi.ClusterConfiguration, _ *kubead
 
 	if kc.config.Authentication.Anonymous.Enabled == nil {
 		kc.config.Authentication.Anonymous.Enabled = utilpointer.BoolPtr(kubeletAuthenticationAnonymousEnabled)
-	} else if *kc.config.Authentication.Anonymous.Enabled != kubeletAuthenticationAnonymousEnabled {
+	} else if *kc.config.Authentication.Anonymous.Enabled {
```

Comment:
i don't see keeping these comparisons with the constants / defaults as a problem.
is this something a static check tool reports as a problem?

Reply:
yes, static check tool  thinks should omit comparison with boolean constant, and it reports this as a problem.
https://staticcheck.io/docs/checks#S1002

Reply:
ok, this cleanup seems fine for the sake of not having follow up PRs that do the same cleanup.